libs/net/entities/Validation/NotificationValidator.cs

```diff
@@ -113,12 +112,12 @@ protected virtual bool ValidateNotification()
     /// <summary>
     /// Validate the notification filter to determine whether it should generate anything.
     /// </summary>
+    /// <param name="filter"></param>
     /// <returns>True if the notification filter has been matched.</returns>
-    protected virtual bool ValidateFilter()
+    protected virtual bool ValidateFilter(NotificationFilter filter)
     {
         if (Notification == null || Content == null) throw new InvalidOperationException("Notification and Content properties cannot be null");
 
-        var filter = JsonSerializer.Deserialize<NotificationFilter>(Notification.Filter);
```

Comment:
Do not remove.  The validator already has access to the notification and thus the filter.  You do not need to pass the filter in.

Reply:
It seems the filter here without _serializationOptions won't work, so if we don't pass the filter in we might still need passing the _serializationOptions in, I guess we could use DI for getting the _serializationOptions here but for some reasons this option seems not working yet.

Reply:
This class is initialized with the Notification object.  It should have the filter.  You'll need to debug the issue and correctly inject the correct serialization options.

Reply:
Updated, thanks